Resources/Prototypes/_DV/Entities/Structures/Machines/smartfridge.yml

```diff
@@ -0,0 +1,118 @@
+- type: entity
+  parent: BaseMachinePowered
+  id: DVSmartFridge
+  name: SmartFridge
+  description: A refrigerated storage unit for keeping items cold and fresh.
+  components:
+  - type: StationAiWhitelist
+  - type: Advertise
+    pack: SmartFridgeAds
+  - type: Speech
+  - type: Appearance
+  - type: Sprite
+    sprite: Structures/Machines/smartfridge.rsi
+    snapCardinals: true
+    layers:
+    - state: smartfridge
+      map: ["enum.StorageVisualLayers.Base"]
+    - state: smartfridge_door
+      map: ["enum.StorageVisualLayers.Door"]
+      shader: unshaded
+  - type: PointLight
+    enabled: false
+    radius: 1.5
+    energy: 1.6
+    color: "#9dc5c9"
+  - type: ContainerContainer
+    containers:
+      entity_storage: !type:Container
+      smart_fridge_inventory: !type:Container
+      smart_fridge_insertion: !type:ContainerSlot
+  - type: LitOnPowered
+  - type: ApcPowerReceiver
+    powerLoad: 200
+  - type: SmartFridge
+    whitelist:
+      components:
+        - FitsInDispenser
+        - Pill
+        - Produce
+        - Seed
+      tags:
+        - PillCanister
+        - Bottle
+        - Syringe
+        - ChemDispensable
+  - type: ActivatableUI
+    key: enum.SmartFridgeUiKey.Key
+  - type: ActivatableUIRequiresPower
+  - type: UserInterface
+    interfaces:
+      enum.SmartFridgeUiKey.Key:
+        type: SmartFridgeBoundUserInterface
+  - type: UseDelay
+    delay: 1
+  - type: AntiRottingContainer
+  - type: ResistLocker
```

Comment:
some of these components are copy pasted and dont matter now that it isnt an EntityStorage

additionally a lot of things are inherited